File: backend/EncryptionUtils.js
const crypto = require('crypto');

class EncryptionUtils {
  // ChatGPT usage: Partial
  static generateKey() {
    // AES-128-ECB uses 16 bytes key
    return crypto.randomBytes(16);
  }
  // ChatGPT usage: Partial
  static encrypt(text, key) {
    const cipher = crypto.createCipheriv('aes-128-ecb', key, null);
    let encrypted = cipher.update(text, 'utf8', 'base64');
    encrypted += cipher.final('base64');
    return encrypted;
  }
  // ChatGPT usage: Partial
  static decrypt(encryptedText, key) {
    const decipher = crypto.createDecipheriv('aes-128-ecb', key, null);
    let decrypted = decipher.update(encryptedText, 'base64', 'utf8');
    decrypted += decipher.final('utf8');
    return decrypted;
  }
}

module.exports = EncryptionUtils;

File: frontend/AndroidCasinoUser/app/src/main/java/com/davidzhang/androidcasinouser/EncryptionUtils.java
package com.davidzhang.androidcasinouser;

import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;

import java.nio.charset.StandardCharsets;

public class EncryptionUtils {

    private static final String ALGORITHM = "AES/ECB/PKCS5Padding";
    private static final byte[] FIXED_KEY = {
            // 16 bytes for AES-128. Example key (in hexadecimal format):
            (byte) 0x01, (byte) 0x23, (byte) 0x45, (byte) 0x67,
            (byte) 0x89, (byte) 0xab, (byte) 0xcd, (byte) 0xef,
            (byte) 0x01, (byte) 0x23, (byte) 0x45, (byte) 0x67,
            (byte) 0x89, (byte) 0xab, (byte) 0xcd, (byte) 0xef
    };

    public static SecretKey generateKey() {
        return new SecretKeySpec(FIXED_KEY, "AES");
    }

    public static byte[] encrypt(String input, SecretKey key) throws Exception {
        Cipher cipher = Cipher.getInstance(ALGORITHM);
        cipher.init(Cipher.ENCRYPT_MODE, key);
        return cipher.doFinal(input.getBytes(StandardCharsets.UTF_8));
    }

    public static String decrypt(byte[] encryptionBytes, SecretKey key) throws Exception {
        Cipher cipher = Cipher.getInstance(ALGORITHM);
        cipher.init(Cipher.DECRYPT_MODE, key);
        byte[] decryptedBytes = cipher.doFinal(encryptionBytes);
        return new String(decryptedBytes, StandardCharsets.UTF_8);
    }
}
